pandas/tests/indexes/test_numeric.py

```diff
@@ -393,6 +393,46 @@ def test_get_loc_missing_nan(self):
             # listlike/non-hashable raises TypeError
             idx.get_loc([np.nan])
 
+    @pytest.mark.parametrize(
+        "vals",
+        [
+            pd.date_range("2016-01-01", periods=3),
+            pd.timedelta_range("1 Day", periods=3),
+        ],
+    )
+    def test_lookups_datetimelike_values(self, vals):
+        # If we have datetime64 or timedelta64 values, make sure they are
+        #  wrappped correctly
+        ser = pd.Series(vals, index=range(3, 6))
+        ser.index = ser.index.astype("float64")
```

Comment:
Will this work on 32-bit version as well? (not a blocker or anything like that, just asking for getting more knowledge)

Reply:
we don't support non-64bit Indexes currently, so yes.

Reply:
Thank you for the explanation @jreback